src/protocol-data-converter.ts

```diff
@@ -20,20 +20,22 @@ function protocolDataToIlpAndCustom ({ protocolData }) {
 
   return {
     protocolMap,
-    ilp: protocolMap.ilp,
-    custom: protocolMap.custom
+    ilp: protocolMap['ilp'],
+    custom: protocolMap['custom']
   }
 }
 
-function ilpAndCustomToProtocolData ({ ilp, custom, protocolMap }) {
+export function ilpAndCustomToProtocolData (data: { ilp?: Buffer, custom?: Object , protocolMap?: Map<string, Buffer | string | Object> }): Array<BtpSubProtocol> {
   const protocolData = []
+  const {ilp, custom, protocolMap} = data
 
   // ILP is always the primary protocol when it's specified
   if (ilp) {
     protocolData.push({
       protocolName: 'ilp',
       contentType: Btp.MIME_APPLICATION_OCTET_STREAM,
-      data: Buffer.from(ilp, 'base64')
+      //TODO JS originally had a Buffer.from(ilp, 'base64')?
```

Comment:
I agree we should get rid of all this from/to base64 conversion now that we're no longer JSON-encoding things anyway, but if we change this now, then other code that uses `ilpAndCustomToProtocolData` needs to be updated to match this change. Where is this function used? Does this module expose it?

Reply:
It is used internally in the plugin and always passes in a Buffer. So the effect is to always call:

`Buffer.from(buffer, 'base64')` which is interpreted in JS as ignoring the 'base64' anyway as far as I can tell